logstash-core/src/main/java/org/logstash/instrument/metrics/gauge/LazyDelegatingGauge.java

```diff
@@ -0,0 +1,94 @@
+ * <p><strong>Intended only for use with Ruby's duck typing, Java consumers should use the specific typed {@link GaugeMetric}</strong></p>
+ */
+public class LazyDelegatingGauge extends AbstractMetric<Object> implements GaugeMetric<Object> {
+
+    protected final String key;
+    protected final List<String> nameSpaces;
+
+    private GaugeMetric lazyMetric;
+
+
+    /**
+     * Constructor - protected so that Ruby may sub class proxy and discourage usage from Java, null initial value
+     *
+     * @param nameSpace The namespace for this metric
+     * @param key       The key <i>(with in the namespace)</i> for this metric
+     */
+    public LazyDelegatingGauge(final List<String> nameSpace, final String key) {
+        this(nameSpace, key, null);
+    }
+
+    /**
+     * Constructor - protected so that Ruby may sub class proxy and discourage usage from Java
+     *
+     * @param nameSpace    The namespace for this metric
+     * @param key          The key <i>(with in the namespace)</i> for this metric
+     * @param initialValue The initial value for this {@link GaugeMetric}, may be null
+     */
+    protected LazyDelegatingGauge(List<String> nameSpace, String key, Object initialValue) {
+        super(nameSpace, key);
+        this.nameSpaces = nameSpace;
+        this.key = key;
+        if (initialValue != null) {
+            wakeMetric(initialValue);
+        }
+    }
+
+    @Override
+    public Object get() {
+        return lazyMetric == null ? null : lazyMetric.get();
+    }
+
+    @Override
+    public MetricType getType() {
+        return lazyMetric == null ? null : lazyMetric.getType();
+    }
+
+    @Override
+    public Object getValue() {
+        return lazyMetric == null ? null : lazyMetric.getValue();
+    }
+
+    @Override
+    public void set(Object value) {
+        if (lazyMetric == null) {
+            wakeMetric(value);
+        } else {
+            lazyMetric.set(value);
+        }
+    }
+
+    /**
+     * Instantiates the metric based on the type used to set this Gauge
+     *
+     * @param value The object used to set this value
+     */
+    private synchronized void wakeMetric(Object value) {
+        if (lazyMetric == null) {
+            //"quack quack"
+            if (value instanceof Number) {
+                lazyMetric = new NumericGauge(nameSpaces, key, (Number) value);
+            } else if (value instanceof String) {
+                lazyMetric = new TextGauge(nameSpaces, key, (String) value);
+            } else if (value instanceof Boolean) {
+                lazyMetric = new BooleanGauge(nameSpaces, key, (Boolean) value);
+            } else if (value instanceof RubyHash) {
+                lazyMetric = new RubyHashGauge(nameSpaces, key, (RubyHash) value);
+            } else {
+                lazyMetric = new UnknownGauge(nameSpaces, key, value);
```

Comment:
I'm wondering now if we should warn on unknown gauge types and ask users to not use them?

Since our long term plan in #7509 is to only support double/text/boolean, it might be good to warn on unknown for now, and once we've refactored the LS internals then move that warning up to rubyhash ?